includes/Core/Authentication/Google_Proxy.php

```diff
@@ -95,6 +95,72 @@ public function get_site_fields() {
 		);
 	}
 
+	/**
+	 * Fetch site fields
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Credentials $credentials Credentials instance.
+	 * @return array|WP_Error The response as an associative array or WP_Error on failure.
+	 */
+	public function fetch_site_fields( Credentials $credentials ) {
+		if ( ! $credentials->has() ) {
+			return null;
+		}
+
+		$creds = $credentials->get();
+
+		$request_args = array(
+			'body' => array(
+				'site_id'     => $creds['oauth2_client_id'],
+				'site_secret' => $creds['oauth2_client_secret'],
+			),
+		);
+
+		$response = wp_remote_post( $this->url( self::OAUTH2_SITE_URI ), $request_args );
+
+		if ( is_wp_error( $response['body'] ) ) {
+			return $response['body'];
+		}
```

Comment:
We need to check the response object itself, not its body:

```suggestion
		if ( is_wp_error( $response ) ) {
			return $response;
		}
```

Reply:
This does not work because the $reponse is a standard array, the WP_Error or other results are returned in the body of the request. In the initial PR I fixed this by testing `is_wp_error($raw_body)`, we can do this or test the `$reponse['body']` let me know which you prefer.

```
var_dump($response);
```

```
array(5) {
  ["headers"]=>
  array(0) {
  }
  ["body"]=>
  object(WP_Error)#3664 (2) {
    ["errors"]=>
    array(1) {
      ["test_message"]=>
      array(1) {
        [0]=>
        string(12) "test_message"
      }
    }
    ["error_data"]=>
    array(0) {
    }
  }
  ["response"]=>
  array(2) {
    ["code"]=>
    int(200)
    ["message"]=>
    string(2) "OK"
  }
  ["cookies"]=>
  array(0) {
  }
  ["http_response"]=>
  NULL
}
```

Reply:
@benbowler see the documentation for the [`wp_remote_post`](https://developer.wordpress.org/reference/functions/wp_remote_post/) function:

```
Return #
(array|WP_Error) The response or WP_Error on failure.
```